### DISCUSSION TITLE
Authorization based on claims from GitLab

### DISCUSSION THREAD

**Discussion by mcrugnola** [OpenIDC/mod_auth_openidc#899] [Q&A]
Hello Team,

I have a configuration with apache+mod_auth_openidc and GitLab as IDP.
I would like to allow access to application only if user has access to the corresponding Gitlab group.
The claims I get from GitLab are the following:

{
"sub":"4",
"sub_legacy":"93440bebf0f7586dc84170520f9acdd710ad0d7829e64da2b8655205b277e1c5",
"name":"puppolo",
"nickname":"puppolo",
"profile":"https://gitlab.myserver.com/puppolo",
"groups":["co-projetcs/ibex","co-projetcs/spider","co-projetcs/culturalgems","co-projetcs/eurocodes","co-templates","co-projetcs/test","co-projetcs/dronee5","co-projetcs","co-templates/ci-cd","co-templates/ptemplates","co-projetcs/dronee5/jrce5_blender_cpvt","co-projetcs/dronee5/livinglab","co-projetcs/mytestapp","co-projetcs/dichi","co-projetcs/refap","co-projetcs/gtcap","co-projetcs/sciart","co-projetcs/refap/SA","co-projetcs/gtcap/test"],
...

I have the information of the group name in an Apache environment variable. 
I wrote a Require statement like this:

Require claim "groups~%{env:groupname}"
and also tried 
Require claim "groups:%{env:groupname}"

but they do not seem to work. Given that groups is an array how is the condition evaluated?

Thanks in  advance

**Comment by zandbelt** [comment]
the 2nd form should work for non-nested string array claims indeed: the array is searched until a matching value is found; if you want to debug what happens, turn `LogLevel auth_openidc:debug` on and check the server log

**Reply by mcrugnola** [reply]
Hello Hans,

thanks, debug info only says authorization result of Require claim "groups:%{env:groupname}": denied:


[Tue Aug 09 16:45:26.945178 2022] [authz_core:debug] [pid 1029298:tid 140612699404032] mod_authz_core.c(820): [client 10.170.129.14:56960] AH01626: authorization result of Require claim "groups:%{env:groupname}": denied
[Tue Aug 09 16:45:26.945188 2022] [authz_core:debug] [pid 1029298:tid 140612699404032] mod_authz_core.c(820): [client 10.170.129.14:56960] AH01626: authorization result of <RequireAny>: denied

**Reply by zandbelt** [reply]
it seems you're using a version that does not support expressions in requirements, which was introduced in version 2.4.2.1

**Reply by mcrugnola** [reply]
Thanks Hans, you are right. I was told we had 2.4.11 installed but it wasn't true.
